### DISCUSSION TITLE
Unable to Start Keycloak

### DISCUSSION THREAD

**Discussion by superdave2** [keycloak/keycloak#26141] [Q&A]
Hi All

New to keycloak and download the latest version of keyclok to install on my ubuntu 20. Followed the basic instructions of from the following link:

https://www.keycloak.org/getting-started/getting-started-zip

However when i attempt to start the service i get :

bin/kc.sh start-dev
bin/kc.sh: 1: eval: /usr/lib/jvm/java-11-openjdk-amd64/bin/java/bin/java: not found

Here is my java version.
openjdk version "17.0.9" 2023-10-17
OpenJDK Runtime Environment (build 17.0.9+9-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 17.0.9+9-Ubuntu-120.04, mixed mode, sharing)

I am not sure if there is a configuration file which is holding an invalid path for the java. 
any feedback is appreciated. 

thanks

**Comment by darius-m** [comment]
Try inspecting your environment using `env` or `printenv`. It's probably a variable set by the environment - either `JAVA` or `JAVA_HOME`. If neither is set, the script should look up the Java installation automatically. You can run a command with a clean environment using `env -i command`, but with some notes, as mentioned in [this stackexchange post](https://unix.stackexchange.com/a/48995) (or search for "linux run command without environment" to get other answers).
It may also be possible that you have a different (incomplete) Java installation on your system that is picked up by the script instead of the one you expect (the script may search for the `java` binary, which can be provided by various packages and set up incorrectly through alternatives - see the `alternatives` or `update-alternatives` commands, not sure which it would be for Ubuntu off the top of my head).
